src/cbc_sdk/platform/alerts.py

```diff
@@ -997,6 +998,132 @@ def set_rows(self, rows):
+            where y=year, w=week, d=day, h=hour, m=minute, s=second.
+
+        For v6 Alerts (backwards compatibility):
+
+        Restricts the alerts that this query is performed on to the specified time range for a given key. Will also set
+        the 'time_range' as in the v7 usage if key is create_time or backend_timestamp. Will be deprecated with v6 alert
+        api
+
+        Args:
+            key (str): The key to use for criteria one of create_time, first_event_time, last_event_time,
+             backend_timestamp, backend_update_timestamp, or last_update_time
+            **kwargs (dict): Used to specify start= for start time, end= for end time, and range= for range. Values are
+            either timestamp ISO 8601 strings or datetime objects for start and end time. For range the time range to
+            execute the result search, ending on the current time. Should be in the form "-2w",
+            where y=year, w=week, d=day, h=hour, m=minute, s=second.
+
+
+        Returns:
+            AlertSearchQuery: This instance.
+
+        Examples:
+            >>> query = api.select(Alert).set_time_range(start="2020-10-20T20:34:07Z")
+            >>> second_query = api.select(Alert).
+            ...     set_time_range(start="2020-10-20T20:34:07Z", end="2020-10-30T20:34:07Z")
+            >>> third_query = api.select(Alert).set_time_range(range='-3d')
+
+        """
+        args_count = args.__len__()
+        time_filter = self._create_valid_time_filter(kwargs)
+        if args_count > 0:
+            key = args[0]
+            if self._is_valid_time_criteria_key(key):
+                self.add_time_criteria(key, **kwargs)
+                if key in ["create_time", "backend_timestamp"]:
+                    self._time_range = time_filter
+                return self
+        else:
+            # everything before this is only for backwards compatibility, once v6 deprecates all the other
+            # checks can be removed
+            self._time_range = {}
+            self._time_range = time_filter
+            return self
+
+    def add_time_criteria(self, key, **kwargs):
+        """
+        Restricts the alerts that this query is performed on to the specified time range for a given key.
+
+        The time may either be specified as a start and end point or as a range.
+
+        Args:
+            key (str): The key to use for criteria one of create_time, first_event_time, last_event_time,
+             backend_timestamp, backend_update_timestamp, or last_update_time
+            **kwargs (dict): Used to specify start= for start time, end= for end time, and range= for range.
+
+        Returns:
+            AlertSearchQuery: This instance.
+        """
+        if self._is_valid_time_criteria_key(key):
+            self._time_filters[key] = self._create_valid_time_filter(kwargs)
+        return self
+
+    def _is_valid_time_criteria_key(self, key):
```

Comment:
Need a different version of this method to be used at line 1043 when supporting the backwards compatibility of set_time_range, and has the fields that are valid for v6 legacy compatibility.

This method with the v7 valid fields should be called in add_time_criteria()